zproject/backends.py

```diff
@@ -1500,6 +1514,17 @@ def auth_complete(self, *args: Any, **kwargs: Any) -> Optional[HttpResponse]:
 
         return result
 
+    @classmethod
+    def validate_idp_for_subdomain(cls, idp_name: str, subdomain: str) -> bool:
+        idp_dict = settings.SOCIAL_AUTH_SAML_ENABLED_IDPS.get(idp_name)
+        if idp_dict is None:  # nocoverage
+            # This is defensive code and should never happen.
+            return False
+        if 'limit_to_subdomains' in idp_dict and subdomain not in idp_dict['limit_to_subdomains']:
+            return False
```

Comment:
I think we might want to make this typo-resistant; maybe we should add a `SAML_REQUIRE_LIMIT_TO_SUBDOMAINS` setting that we can set for zulipchat.com and rejects-and-logs any declared IDP that doesn't have this property?

Reply:
Added a commit for this